compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1831,6 +1831,7 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                 self.suggest_as_ref_where_appropriate(span, &exp_found, diag);
                 self.suggest_accessing_field_where_appropriate(cause, &exp_found, diag);
                 self.suggest_await_on_expect_found(cause, span, &exp_found, diag);
+                self.suggest_function_pointers(cause, span, &exp_found, diag);
```

Comment:
So in addition to `exp_found`, there are another set of `exp_p` and `found_p` tys that you may be able to thread into `suggest_function_pointers`. I *think* that might give us a better error message for the case where we have `&{fn item}` vs `&{fn item}`.

Reply:
Could you explain a bit on how `exp_p` and `found_p` might help us with error messages?
I checked the values of `exp_p` and `found_p` using the tests in `src/test/ui/fn/fn-pointer-mismatch.rs` and they both evaluate to `None`. I'm still learning about the significance of these two variables.

Reply:
Hm... maybe I'm mistaken. I thought these would be pointing at the deepest part of the type mismatch, e.g. given

`Arc<Box<fn()>>` and `Arc<Box<{fn item}>>`, somewhere else in this logic I thought we had something that peeled these down to `fn()` and `{fn item}` -- if we can pass those in, we'd get better notes.